### PR TITLE
FINERACT-1552: Upgrade fineract-client to OpenAPI generator 5.4.0

### PR DESCRIPTION
**Pull request by vidakovic**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [x] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

**Comment by vorburger** [comment]
@vidakovic how about client tests for this new UploadRequest and DocumentUploadRequest?

**Comment by vidakovic** [comment]
@vorburger ... sure... I've used the LoanApiResource to test if everything is still working with the "normal" integration tests... I can see to replicate that... but apparently there's still an error to be solved. But overall I think this strategy should work.

**Comment by vidakovic** [comment]
Hmmm... that's odd... JSON syntax errors in places where I didn't change a thing... have to look into this a bit more...

File: integration-tests/src/test/java/org/apache/fineract/integrationtests/client/ClientTest.java [hunk 1]
```diff
@@ -65,7 +65,8 @@ Long create() {
                 fineract().clients.retrieveAll21(null, null, null, null, null, null, null, null, 0, 1, null, null, false));
         if (clients.getTotalFilteredRecords() > 0) {
             // TODO rm long cast, see https://issues.apache.org/jira/browse/FINERACT-1230
-            return Optional.of((long) clients.getPageItems().get(0).getId());
+            return clients.getPageItems().stream().findFirst().map(item -> item.getId().longValue());
+            // return Optional.of((long) clients.getPageItems().stream().findFirst().get().getId());
```
Comment (galovics) [inline integration-tests/src/test/java/org/apache/fineract/integrationtests/client/ClientTest.java]: I guess we wanna uncomment this or remove the commented code.

Comment (vidakovic) [reply integration-tests/src/test/java/org/apache/fineract/integrationtests/client/ClientTest.java]: Yup... one more round to demonstrate that the uploads are (well, at least one unit test) working with the new client (as Michael suggested... this was the initial blocker why we couldn't upgrade)... and with that I'll cleanup this here too.

:+1:

**Comment by vidakovic** [comment]
@vorburger ... just saw that you already implemented upload tests with image and document client API... the tests pass, so I think we are good here... of course we could always use more tests... have some separate test related PRs where I can circle back to that later.

**Comment by vidakovic** [comment]
@vorburger I've changed the typs from Integer to Long in the Swagger descriptor. I agree with you that we should "standardize" on Long for IDs. This specific change was easy and should have no further side effects (other than it generates a nicer API client).  For the JSON de-/serialization this change doesn't matter anyway.

**Review by galovics — APPROVED** [review]

**Pull request merged**